internal/container/container.go

```diff
@@ -40,6 +40,27 @@ func DoesExist(ctx context.Context, name string) (bool, error) {
 	return len(containers) > 0, nil
 }
 
+// Pull pulls the given image.
+func Pull(ctx context.Context, image string) error {
+	cli, err := dockerClient.Get(ctx)
+	if err != nil {
+		return err
+	}
+
+	reader, err := cli.ImagePull(ctx, image, types.ImagePullOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to pull %s image: %w", image, err)
+	}
+	defer reader.Close()
+
+	_, err = io.Copy(os.Stderr, reader)
```

Comment:
Grab a logger from the context with `logger := log.FromContext(ctx)` and use `logger.WriterAt(log.Info)` to get a writer that will go through the logger.

Reply:
From memory the output is designed for a terminal though, but I think there's an option to modify it...maybe? Should be fine for now.

Reply:
Works great, still messy but at least it's going through the logger.